driver-core/src/main/com/mongodb/TransactionOptions.java

```diff
@@ -42,6 +42,7 @@ public final class TransactionOptions {
     private final WriteConcern writeConcern;
     private final ReadPreference readPreference;
     private final Long maxCommitTimeMS;
+    private final Long timeoutMS;
```

Comment:
The specification seems ambiguous about the use of `timeoutMS`, particularly whether it's intended to apply solely to withTransaction. This [discussion in the specifications pull request](https://github.com/mongodb/specifications/pull/881/files#r543057367) suggests an initial intention for timeoutMS to be set/unset for withTransaction.

However, when timeoutMS is specified in `TransactionOptions`, it appears to impact the `startTransaction` method as well, not just `withTransaction`. Do we need to seek further clarification on this?

Reply:
I think we follow the spec see: [TransactionOptions](https://github.com/mongodb/specifications/blob/master/source/transactions-convenient-api/transactions-convenient-api.rst#user-content-transactionoptions) which is described as:

> Options for {{ClientSession.startTransaction}}, as defined in the [Transactions](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.md) specification. The structure of these options MAY vary across drivers (e.g. dictionary, typed class).

Reply:
Apologies, yes we do deviate from spec slightly but not really with the spirit of the [timeoutMS cannot be overridden for startSession calls]:(https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.md#timeoutms-cannot-be-overridden-for-startsession-calls) 

As we already share {{TransactionOptions}} with both APIs I think we can do this, without much harm. It does add a potential configuration knob but only if a user wishes to use it.